### ISSUE TITLE
noqa is ignored

### ISSUE THREAD

**Issue by romulus-ai** [ansible/ansible-lint#2488]
##### Summary

a # noqa comment is ignored by ansible-lint.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.12.8]
  config file = /root/.ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /root/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /root/.local/bin/ansible
  python version = 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
  jinja version = 3.1.2
  libyaml = False

ansible-lint --version
ansible-lint 6.0.2 using ansible 2.12.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

running in docker ubuntu 22.04 image

##### STEPS TO REPRODUCE
Have a task like this in your playbook:

```yaml
- name: "update system if activated"
  ansible.builtin.yum:
    name: '*'
    update_cache: true
    state: latest # noqa package-latest
```

Lint the playbook with "ansible-lint /path/to/playbook.yaml"

```console (paste below)
### LINTING PLAYBOOK playbook.yaml ###
WARNING: PATH altered to include /usr/bin
WARNING  Listing 1 violation(s) that are fatal
package-latest: Package installs should not use latest.
../../../../../../../../root/.ansible/roles/common/tasks/default.yaml:144 Task/Handler: update system if activated
You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - package-latest  # Package installs should not use latest.
Finished with 1 failure(s), 0 warning(s) on 25 files.
```

Skipping the check with "-x package-latest" works, however I plan to use the role in a bunch of other playbooks and CI pipelines, so it would be much nicer to just deactivate the check with noqa-comment. Especially, since for there is another way for apt, I want to have the check for all other tasks except this one.

##### Desired Behavior

The comment should be respected and not error should be thrown

##### Actual Behavior

see steps to reproduce.

**Comment by hardoverflow** [comment]
With `ansible-lint==6.6.1`
```bash
WARNING  Listing 1 violation(s) that are fatal
test.yml:9: yaml: too few spaces before comment (yaml[comments])
You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - yaml[comments]  # Violations reported by yamllint.

Finished with 1 failure(s), 0 warning(s) on 1 files.
```
Your problem seems to be a missing whitespace in the last line. Here is the correct line:
```
- name: "update system if activated"
  ansible.builtin.yum:
    name: '*'
    update_cache: true
    state: latest  # noqa package-latest
```

**Comment by romulus-ai** [comment]
Hey @hardoverflow,
Thanks for answering! Tested it with the line you suggested. So you mean the two spaces between "latest" and "#"? I added two now but the problem remains.

**Comment by ssbarnea** [comment]
Please do not file bugs for old versions. In fact we recommend checking with main branch first if possible.
